examples/stable_diffusion_xl/gm/helpers.py

```diff
@@ -644,16 +649,45 @@ def save_checkpoint(model, path, only_save_lora=False):
         else:
             ckpt.append({"name": n, "data": p})
 
+    delete_checkpoint(ckpt_queue, max_num_ckpt, only_save_lora)
+
     if not only_save_lora:
         ms.save_checkpoint(ckpt, path)
         print(f"save checkpoint to {path}")
 
     if len(ckpt_lora) > 0:
-        path_lora = path[: -len(".ckpt")] + "_lora.ckpt"
+        path_lora = _build_lora_ckpt_path(path)
         ms.save_checkpoint(ckpt_lora, path_lora)
         print(f"save lora checkpoint to {path_lora}")
 
 
+def delete_checkpoint(ckpt_queue, max_num_ckpt, only_save_lora):
+    """
+    Only keep the latest `max_num_ckpt` ckpts while training. If max_num_ckpt == 0, keep all ckpts.
+    """
+    if max_num_ckpt > 0 and len(ckpt_queue) >= max_num_ckpt:
+        del_ckpt = ckpt_queue.pop(0)
+        del_ckpt_lora = _build_lora_ckpt_path(del_ckpt)
+        if only_save_lora:
+            del_ckpts = [del_ckpt_lora]
+        else:
+            del_ckpts = [del_ckpt, del_ckpt_lora]
+
+        for to_del in del_ckpts:
+            if os.path.isfile(to_del):
+                try:
+                    os.chmod(to_del, stat.S_IWRITE)
+                    os.remove(to_del)
+                    print(
```

Comment:
Use a logger to print out messages. And this message could be a debug message as it's not important.

Reply:
Updated